sdk/python/feast/go_server.py

```diff
@@ -249,96 +216,71 @@ def get_online_features(
 
     def _start_go_server_use_thread(self):
 
-        self._go_server_background_thread = GoServerBackgroundThread(
-            "GoServerBackgroundThread",
-            self._shared_connection,
-            self._go_server_started,
+        self._go_server_background_thread = GoServerMonitorThread(
+            "GoServerMonitorThread", self._shared_connection, self._go_server_started
         )
         self._go_server_background_thread.start()
-        atexit.register(lambda: self._go_server_background_thread.stop_go_server())
+        atexit.register(lambda: self._go_server_background_thread.stop())
         signal.signal(
-            signal.SIGTERM,
-            lambda sig, frame: self._go_server_background_thread.stop_go_server(),
+            signal.SIGTERM, lambda sig, frame: self._go_server_background_thread.stop(),
         )
         signal.signal(
-            signal.SIGINT,
-            lambda sig, frame: self._go_server_background_thread.stop_go_server(),
+            signal.SIGINT, lambda sig, frame: self._go_server_background_thread.stop(),
         )
 
         # Wait for go server subprocess to start for the first time before returning
         self._go_server_started.wait()
 
-    def _start_go_server(self):
-        if self._shared_connection.is_process_alive():
-            self._shared_connection.kill_process()
-
-        self._shared_connection.connect()
-        atexit.register(lambda: self._shared_connection.kill_process())
-        signal.signal(
-            signal.SIGTERM, lambda sig, frame: self._shared_connection.kill_process()
-        )
-        signal.signal(
-            signal.SIGINT, lambda sig, frame: self._shared_connection.kill_process()
-        )
-
     def kill_go_server_explicitly(self):
-        if self._check_use_thread():
-            self._go_server_background_thread.stop_go_server()
-        else:
-            self._shared_connection.kill_process()
+        self._go_server_background_thread.stop()
 
 
-# https://www.geeksforgeeks.org/python-different-ways-to-kill-a-thread/
-class GoServerBackgroundThread(threading.Thread):
+class GoServerMonitorThread(threading.Thread):
     def __init__(
         self,
         name: str,
         shared_connection: GoServerConnection,
-        go_server_started: threading.Event,
+        go_server_first_started: threading.Event,
     ):
         threading.Thread.__init__(self)
-        self.name = name
+        self.setName(name)
         self._shared_connection = shared_connection
-        self._go_server_started = go_server_started
+        self._is_cancelled = False
+        self.setDaemon(True)
```

Comment:
similar to `name`, `daemon` should be used directly as a property: https://docs.python.org/3.7/library/threading.html#threading.Thread.daemon